apps/meteor/tests/end-to-end/api/09-rooms.js

```diff
@@ -239,6 +251,34 @@ describe('[Rooms]', function () {
 			await request.get(fileOldUrl).set(credentials).expect('Content-Type', 'image/png').expect(200);
 		});
 
+		it('should be able to get the file if not member but can access room if setting allows', async () => {
+			await updateSetting('FileUpload_Restrict_to_room_members', false);
+			await updateSetting('FileUpload_Restrict_to_users_who_can_access_room', true);
+
+			await request.get(fileNewUrl).set(credentials).expect('Content-Type', 'image/png').expect(200);
```

Comment:
By using `credentials` the user is a room member, isn't it? Following the test description we should use `userCredentials` instead if we want a not member user, right?